src/main/java/com/github/manolo8/darkbot/extensions/features/FeatureRegistry.java

```diff
@@ -65,12 +73,32 @@ public void beforeLoad() {
     @Override
     public void afterLoad() {
         pluginHandler.LOADED_PLUGINS.forEach(pl ->
-                Arrays.stream(pl.getDefinition().features)
+                getFeaturesList(pl)
                         .forEach(feature -> registerPluginFeature(pl, feature)));
         pluginHandler.LOADED_PLUGINS.sort((p1, p2) -> Boolean.compare(isDisabled(p1), isDisabled(p2)));
         registryHandler.update();
     }
 
+    private List<String> getFeaturesList(Plugin plugin) {
+        if (plugin.getDefinition().features.length != 0) return Arrays.asList(plugin.getDefinition().features);
+        URLClassLoader loader = new URLClassLoader(new URL[]{plugin.getJar()});
+        List<String> features = new LinkedList<>();
+        try (JarFile jarFile = new JarFile(plugin.getFile())) {
+            Enumeration<JarEntry> e = jarFile.entries();
+            while (e.hasMoreElements()) {
+                JarEntry jarEntry = e.nextElement();
+                if (jarEntry.getName().endsWith(".class")) {
+                    String className = jarEntry.getName().replace("/", ".").replace(".class", "");
+                    if(loader.loadClass(className).isAnnotationPresent(RegisterFeature.class))
```

Comment:
```suggestion
                    if (loader.loadClass(className).isAnnotationPresent(RegisterFeature.class))
```
small formatting change